cmd/mdatagen/main_test.go

```diff
@@ -620,6 +621,132 @@ func LeveledMeter(settings component.TelemetrySettings, level configtelemetry.Le
 func Tracer(settings component.TelemetrySettings) trace.Tracer {
 	return settings.TracerProvider.Tracer("")
 }
+`,
+		},
+		{
+			name: "foo component with internal telemetry",
+			md: metadata{
+				Type: "foo",
+				Status: &Status{
+					Stability: map[component.StabilityLevel][]string{
+						component.StabilityLevelAlpha: {"metrics"},
+					},
+					Distributions: []string{"contrib"},
+					Class:         "receiver",
+				},
+				Telemetry: telemetry{
+					Metrics: map[metricName]metric{
+						"metric1": {
+							Enabled: true,
+							Sum: &sum{
+								Async:           false,
+								Mono:            Mono{true},
+								MetricValueType: MetricValueType{ValueType: pmetric.NumberDataPointValueTypeInt},
```

Comment:
Could we add another case where the type is `Histogram`, since that's the other special case in the generated code?